src/io/flutter/server/vmService/VMServiceManager.java

```diff
@@ -245,6 +248,16 @@ private void addServiceExtension(String name) {
       else if (!stream.getValue()) {
         stream.setValue(true);
       }
+
+      if (serviceExtensionState.get(name) != null && serviceExtensionState.get(name).getValue()) {
+        // We should not call the service extension for the follwing extension.
```

Comment:
why doesn't show work?

I would have thought you could go through the same code path just with this code added as a bonus if the actual button was clicked. For a hot restart, bringing the emulator to the front is actually a bug more than a feature as I can't think of how it actually makes sense. 

      // If toggling inspect mode on, bring all devices to the foreground.
      // TODO(jacobr): consider only bringing the device for the currently open inspector TAB.
      if (isSelected()) {
        final FlutterDevice device = app.device();
        if (device != null) {
          device.bringToFront();
        }
      }

Reply:
I didn't actually write this code, just copied it over from where it used to be in FlutterView.

Reply:
Same for all the actions in the new files.